js/braziliex.js

```diff
@@ -0,0 +1,425 @@
+                    'taker': 0.005,
+                },
+            },
+            'precision': {
+                'amount': 8,
+                'price': 8,
+            },
+        });
+    }
+
+    async fetchCurrencies (params = {}) {
+        let currencies = await this.publicGetCurrencies (params);
+        let ids = Object.keys (currencies);
+        let result = {};
+        for (let i = 0; i < ids.length; i++) {
+            let id = ids[i];
+            let currency = currencies[id];
+            let precision = currency['decimal'];
+            let code = this.commonCurrencyCode (id.toUpperCase ());
+            let active = currency['active'] == 1;
+            let status = 'ok';
+            if (currency['under_maintenance'] != 0) {
+                active = false;
+                status = 'maintenance';
+            }
+            let canWithdraw = currency['is_withdrawal_active'] == 1;
+            let canDeposit = currency['is_deposit_active'] == 1;
+            result[code] = {
+                'id': id,
+                'code': code,
+                'name': currency['name'],
+                'active': active,
+                'status': status,
+                'precision': precision,
+                'wallet': {
+                    'address': undefined,
+                    'extra': undefined,
+                    'withdraw': {
+                        'active': canWithdraw,
+                        'fee': currency['txWithdrawalFee'],
+                    },
+                    'deposit': {
+                        'active': canDeposit,
+                        'fee': currency['txDepositFee'],
+                    },
+                },
+                'limits': {
+                    'amount': {
+                        'min': currency['minAmountTrade'],
+                        'max': Math.pow (10, precision),
+                    },
+                    'price': {
+                        'min': Math.pow (10, -precision),
+                        'max': Math.pow (10, precision),
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'withdraw': {
+                        'min': currency['MinWithdrawal'],
+                        'max': Math.pow (10, precision),
+                    },
+                    'deposit': {
+                        'min': currency['minDeposit'],
+                        'max': undefined,
+                    },
+                },
+                'info': currency,
+            };
+        }
+        return result;
+    }
+
+    async fetchMarkets () {
+        let markets = await this.publicGetTicker ();
+        let ids = Object.keys (markets);
+        let result = [];
+        for (let i = 0; i < ids.length; i++) {
+            let id = ids[i];
+            let market = markets[id];
+            let [ base, quote ] = id.toUpperCase ().split ('_');
```

Comment:
This is the last one, if you can split the rhs chain in two lines, it will satisfy the PHP as well. Sorry for this, I know it can be frustrating.